faust2 won't build anymore

a few weeks ago I was able to successfully install faust2 (nixos.faust), but today it looks like its broken.  I'm on the unstable channel.  

```
[bburdette@nixosthe1:~/code/faustthings/faust-2.0.a41]$ nix-env -iA nixos.faust
installing ‘faust-2.0-a41’
these derivations will be built:
  /nix/store/fwy9lfqhfq6b7kcyp26pf93b4xi12552-faust-2.0-a41.drv
building path(s) ‘/nix/store/5rsqv2c2a3nsrff1c1lgw69bcd30v6d0-faust-2.0-a41’
unpacking sources
unpacking source archive /nix/store/3zf5h1w7kxlrlh3dkwxnbyiglw4cicg7-faust-2.0.a41.tgz
source root is faust-2.0.a41
setting SOURCE_DATE_EPOCH to timestamp 1444409208 of file faust-2.0.a41/windows/template.props
patching sources
configuring
no configure script, doing nothing
building
make flags: SHELL=/nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash  prefix=/nix/store/5rsqv2c2a3nsrff1c1lgw69bcd30v6d0-faust-2.0-a41 LLVM_CONFIG='/nix/store/m5b0r6lznlqmv9dzbc2vhyw1v3n8vvk5-llvm-3.7.1/bin/llvm-config' world   
/nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash: git: command not found
/nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash: git: command not found
/nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash: git: command not found
/nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash: git: command not found
/nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash: git: command not found
/nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash: git: command not found
make -C compiler -f Makefile.unix prefix=/nix/store/5rsqv2c2a3nsrff1c1lgw69bcd30v6d0-faust-2.0-a41
make[1]: Entering directory '/tmp/nix-build-faust-2.0-a41.drv-0/faust-2.0.a41/compiler'
g++ -g -Wall -Wuninitialized  -Wno-parentheses -I. -Iboxes -Ierrors -Ievaluate -Inormalize -Iparser -Ipropagate -Iparallelize -Isignals -Itlib -Idraw -Idraw/device -Idraw/schema -Iextended -Ipatternmatcher -Idocumentator -Iutils -I../libraries -Igenerator -Igenerator/asmjs -Igenerator/c -Igenerator/cpp -Igenerator/java -Igenerator/js -Igenerator/llvm -Igenerator/fir -Igenerator/wasm -DINSTALL_PREFIX='"/nix/store/5rsqv2c2a3nsrff1c1lgw69bcd30v6d0-faust-2.0-a41"' -O1 -fPIC -Wall -Wuninitialized -fvisibility=hidden -Wno-overloaded-virtual -Wno-parentheses -I. -Iboxes -Ierrors -Ievaluate -Inormalize -Iparser -Ipropagate -Iparallelize -Isignals -Itlib -Idraw -Idraw/device -Idraw/schema -Iextended -Ipatternmatcher -Idocumentator -Iutils -I../libraries -Igenerator -Igenerator/asmjs -Igenerator/c -Igenerator/cpp -Igenerator/java -Igenerator/js -Igenerator/llvm -Igenerator/fir -Igenerator/wasm -DINSTALL_PREFIX='"/nix/store/5rsqv2c2a3nsrff1c1lgw69bcd30v6d0-faust-2.0-a41"'  -I`'/nix/store/m5b0r6lznlqmv9dzbc2vhyw1v3n8vvk5-llvm-3.7.1/bin/llvm-config' --includedir` -I../architecture -D3.7.1 -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS    -c -o documentator/doc_sharing.o documentator/doc_sharing.cpp
<command-line>:0:1: error: macro names must be identifiers
<builtin>: recipe for target 'documentator/doc_sharing.o' failed
make[1]: *** [documentator/doc_sharing.o] Error 1
make[1]: Leaving directory '/tmp/nix-build-faust-2.0-a41.drv-0/faust-2.0.a41/compiler'
Makefile:44: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/fwy9lfqhfq6b7kcyp26pf93b4xi12552-faust-2.0-a41.drv’ failed with exit code 2
error: build of ‘/nix/store/fwy9lfqhfq6b7kcyp26pf93b4xi12552-faust-2.0-a41.drv’ failed
```

--- a/pkgs/applications/audio/faust/faust2.nix
+++ b/pkgs/applications/audio/faust/faust2.nix
@@ -37,7 +37,8 @@ let
 
     inherit src;
 
-    buildInputs = [ makeWrapper llvm emscripten openssl libsndfile pkgconfig libmicrohttpd vim ];
+    nativeBuildInputs = [ makeWrapper pkgconfig vim ];
+    buildInputs = [ llvm emscripten openssl libsndfile libmicrohttpd ];
 
 
     passthru = {
@@ -53,6 +54,20 @@ let
       # correct system.
       unset system
       sed -e "232s/LLVM_STATIC_LIBS/LLVMLIBS/" -i compiler/Makefile.unix
+
+      # The makefile sets LLVM_<version> depending on the current llvm
+      # version, but the detection code is quite brittle.
+      #
+      # Failing to properly detect the llvm version means that the macro
+      # LLVM_VERSION ends up being the raw output of `llvm-config --version`, while
+      # the code assumes that it's set to a symbol like `LLVM_35`.  Two problems result:
+      # * <command-line>:0:1: error: macro names must be identifiers.; and
+      # * a bunch of undefined reference errors due to conditional definitions relying on
+      #   LLVM_XY being defined.
+      #
+      # For now, fix this by 1) pinning the llvm version; 2) manually setting LLVM_VERSION
+      # to something the makefile will recognize.
+      sed '52iLLVM_VERSION=3.7.0' -i compiler/Makefile.unix
     '';
 
     # Remove most faust2appl scripts since they won't run properly

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -15895,7 +15895,9 @@ let
 
   faust1 = callPackage ../applications/audio/faust/faust1.nix { };
 
-  faust2 = callPackage ../applications/audio/faust/faust2.nix { };
+  faust2 = callPackage ../applications/audio/faust/faust2.nix {
+    llvm = llvm_37;
+  };
 
   faust2alqt = callPackage ../applications/audio/faust/faust2alqt.nix { };
 

